### DISCUSSION TITLE
Updating existing Laravel CRUD to Livewire

### DISCUSSION THREAD

**Discussion by longestdrive** [livewire/livewire#8642] [1. Help]
Hi
I'm jumping in to using Livewire for the first time and attempting to migrate an app I have to include some Livewire functionality.

I'm trying to get my head around the concept of CRUD and found plenty of tutorials to work through from the concept of a table to then have create, update and delete features - all good there. But what about the Read element?

Edit and Create form appears to be on the same "page" and that's ok - but is it good practice to include the read elements on that page to say read/preview an article? Should I create a separate page for the read/show element?

Could the read page include the elements to edit etc but hidden until say a button is clicked? I could do the read as a modal from the table but may prefer a page view

Are there any examples I could take a look at?

I'm worried I'm not taking full advantage of Livewire if I go down a familiar route that I already have working with CRUD

Thank you :)

**Comment by joshhanley** [comment]
@longestdrive it all comes down to personal preference. Some people prefer to have a table view and show a modal for view or edit. Others (like me) prefer to redirect to an edit page or a view page.

So you should do whatever feels best for your app.

If it feels a lot like just doing CRUD in Laravel, well that's because it basically is just Laravel. That's the beauty of it 😁

But if you need some extra interactivity, that you typically might reach for javascript to do, you can easily use Livewire and Alpine to do that instead. Maybe something like a dynamic row on a form, or a confirmation modal, or something like that.

In terms of page split up, I personally prefer 3 different pages, view, create, and edit all separate. Then I make use of Blade components and Livewire form objects, to share common structure between them.

But if you are happy with them all being the same page with some flags for view vs. edit, then go for it.

Hope this helps!

**Comment by longestdrive** [comment]
Thank you - I think I'm getting stuck on the thoughts that using livewire has to be different to what I do - but your comments make sense :)
